x-pack/plugins/observability_solution/logs_data_access/server/services/get_logs_rates_service/index.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { estypes } from '@elastic/elasticsearch';
+
+interface Params {
+  esClient: ElasticsearchClient;
+  serviceNames: string[];
+  identifyingMetadata: string;
+  timeFrom: number;
+  timeTo: number;
+}
+
+interface LogErrorsAggregation extends estypes.AggregationsStringRareTermsBucketKeys {
+  logErrors: estypes.AggregationsTermsAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+}
+
+interface LogRateQueryAggregation {
+  services: estypes.AggregationsTermsAggregateBase<LogErrorsAggregation>;
+}
+
+export function createGetLogsRatesService() {
+  return async ({ esClient, identifyingMetadata, serviceNames, timeFrom, timeTo }: Params) => {
+    const esResponse = await esClient.search({
```

Comment:
Would it make sense to implement this directly in ES|QL?

Reply:
Can you clarify why?

Reply:
Instead of using the Elasticsearch DSL for the query, use ES|QL syntax for the query. @dgieselaar You might help to chime how this would look like / change the implementation here.

Reply:
I want to understand why I should use ES|QL instead of Elasticsearch DSL. Is the idea to drop the usage of the Elasticsearch DSL?

Reply:
FWIW, the equivalent ES|QL query should be something like:

```
FROM logs-*-*
| WHERE log.level IS NOT NULL AND ${identifyingMetadata} IN (${serviceNames.join(', ')}) AND @timestamp >= ${timeFrom} AND @timestamp <= ${timeTo}
| STATS COUNT(*) BY ${identifyingMetadata}, log.level
| WHERE log.level == "error"
```

Reply:
@cauemarcondes ES|QL just went GA and the expectation is that ES|QL will be faster for many queries and reduce the maintenance as the queries are simpler to read / understand. We want to start adopt it more in our own internal code.

Reply:
We haven't discussed the use of ES|QL in this project yet. IMO we should be using it for absolutely everything but we haven't received any direct guidance asking us to start using it here.

We need to discuss as a team to make sure everyone understands the priority and how to properly use ES|QL in the queries we use in the UI.

Other than the AI assistant, is ES|QL being used anywhere else in Observability?

Reply:
I'm working on a doc to provide this guidance and share it more widely soon. In a nutshell, if you can, start using ES|QL and ping me on it.

Reply:
Ah, that must by how Discover sends their queries